### PR TITLE
I2C support added to Kinesis Advantage Stapelberg PCB by freeing pins D0 and D1

### PR DESCRIPTION
**Pull request by mdbijl**
## Description

The Teensy 2.0+ supports I2C on pins D0 and D1. Unfortunately the Stapelberg controller uses these pins to map key columns. In this configuration I moved the first two column pins from D0 and D1 to E6 and E7 respectively.

I added it as a separate directory (besides the existing 'kinses/stapelberg') so I2C can be added by changing the make command (eg `make kinesis/stapelberg_i2c:default:teensy` vs `make kinesis/stapelberg:default:teensy`)

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+	// put your per-action keyboard code here
+	// runs for every action, just before processing by the firmware
+
+	return process_record_user(keycode, record);
+}
+
+void led_init_ports() {
```
Comment (yanfali) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: delete these if you aren't using them

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  // * Set our LED pins as output
+  // DDRF |= (1<<0); // Keypad LED
+  // DDRF |= (1<<1); // ScrLock LED
+  // DDRF |= (1<<2); // NumLock LED
+  // DDRF |= (1<<3); // CapsLock LED
+}
+
+void led_set_kb(uint8_t usb_led) {
```
Comment (yanfali) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: ditto

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
+#ifndef KINESIS_STAPELBERG_H
+#define KINESIS_STAPELBERG_H
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+
+void kinesis_led_1_on(void);
```
Comment (yanfali) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.h]: why define functions that don't exist?

File: keyboards/kinesis/stapelberg_i2c/config.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#ifndef STAPELBERG_CONFIG_H
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/kinesis/stapelberg_i2c/config.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+#define STAPELBERG_CONFIG_H
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/config.h]:
```suggestion
```

File: keyboards/kinesis/stapelberg_i2c/config.h [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+#define IGNORE_MOD_TAP_INTERRUPT
+
+#endif
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/config.h]:
```suggestion
```

File: keyboards/kinesis/stapelberg_i2c/config.h [hunk 7]
```diff
@@ -0,0 +1,37 @@
+#ifndef STAPELBERG_CONFIG_H
+#define STAPELBERG_CONFIG_H
+
+#include "../config.h"
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/config.h]:
```suggestion
```

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c [hunk 8]
```diff
@@ -0,0 +1,61 @@
+#include "stapelberg_i2c.h"
+
+void matrix_init_kb(void) {
+	// put your keyboard start-up code here
+	// runs once when the firmware starts up
+    // * Set our LED pins as output
+    DDRF |= (1<<0); // Keypad LED
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: Could these be updated to use the GPIO Controls? 
```suggestion
    setPinOutput(F0); // Keypad LED
```

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    DDRF |= (1<<1); // ScrLock LED
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: Could these be updated to use the GPIO Controls? 
```suggestion
    setPinOutput(F1); // ScrLock LED
```

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    DDRF |= (1<<2); // NumLock LED
+    DDRF |= (1<<3); // CapsLock LED
```
Comment (drashna) [inline keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: Could these be updated to use the GPIO Controls? 
```suggestion
    setPinOutput(F3); // CapsLock LED
```

File: keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c (hunk 2 above)
Comment (drashna) [reply keyboards/kinesis/stapelberg_i2c/stapelberg_i2c.c]: That or use the newer version of this function that can be controlled by users.
https://docs.qmk.fm/#/custom_quantum_functions?id=example-led_update_kb-implementation

**Pull request closed without merging**